lightautoml/ml_algo/boost_xgb.py

```diff
@@ -0,0 +1,295 @@
+        # elif rows_num > 10000:
+        #     suggested_params["num_leaves"] = 32 if task == "reg" else 64
+        #     suggested_params["reg_alpha"] = 0.5 if task == "reg" else 0.2
+        # elif rows_num > 5000:
+        #     suggested_params["num_leaves"] = 24 if task == "reg" else 32
+        #     suggested_params["reg_alpha"] = 0.5 if task == "reg" else 0.5
+        # else:
+        #     suggested_params["num_leaves"] = 16 if task == "reg" else 16
+        #     suggested_params["reg_alpha"] = 1 if task == "reg" else 1
+
+        # suggested_params["learning_rate"] = init_lr
+        # suggested_params["num_trees"] = ntrees
+        # suggested_params["early_stopping_rounds"] = es
+
+        return suggested_params
+
+    def _get_default_search_spaces(self, suggested_params: Dict, estimated_n_trials: int) -> Dict:
+        """Sample hyperparameters from suggested.
+
+        Args:
+            suggested_params: Dict with parameters.
+            estimated_n_trials: Maximum number of hyperparameter estimations.
+
+        Returns:
+            dict with sampled hyperparameters.
+
+        """
+        optimization_search_space = {}
+
+        optimization_search_space["colsample_bytree"] = Choice(options=[0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0])
+        optimization_search_space["subsample"] = Choice(options=[0.4, 0.5, 0.6, 0.7, 0.8, 1.0])
+        optimization_search_space["max_depth"] = Choice(options=[5, 7, 9, 11, 13, 15, 17])
+        optimization_search_space["learning_rate"] = Choice(options=[0.008, 0.01, 0.012, 0.014, 0.016, 0.018, 0.02])
+
+        if estimated_n_trials > 30:
+            optimization_search_space["min_child_weight"] = Uniform(low=1, high=300, q=1, log=False)
+
+            optimization_search_space["reg_alpha"] = Uniform(
+                low=1e-3,
+                high=10.0,
+                log=True,
+            )
+            optimization_search_space["reg_lambda"] = Uniform(
+                low=1e-3,
+                high=10.0,
+                log=True,
+            )
+
+        return optimization_search_space
+
+    def fit_predict_single_fold(
+        self, train: TabularDataset, valid: TabularDataset
+    ) -> Tuple[xgboost.Booster, np.ndarray]:
+        """Implements training and prediction on single fold.
+
+        Args:
+            train: Train Dataset.
+            valid: Validation Dataset.
+
+        Returns:
+            Tuple (model, predicted_values)
+
+        """
+        (
+            params,
+            num_trees,
+            early_stopping_rounds,
+            verbose_eval,
+            fobj,
+            feval,
+        ) = self._infer_params()
+
+        train_target, train_weight = self.task.losses["xgb"].fw_func(train.target, train.weights)
+        valid_target, valid_weight = self.task.losses["xgb"].fw_func(valid.target, valid.weights)
+
+        dtrain = xgboost.DMatrix(train.data, label=train_target, weight=train_weight)
```

Comment:
А при обучении учитываются категориальные данные?

Reply:
@aPovidlo yep, for sure categorical features are used in our Lightautoml because we have not only models in our pipelines but also feature engineering/preparation

Reply:
@alexmryzhkov Are the categorical features encoded? One-hot or label?